src/dbnode/storage/namespace/schema.go

```diff
@@ -113,6 +112,19 @@ func (sr *schemaRegistry) Equal(o SchemaRegistry) bool {
 	return true
 }
 
+func (sr *schemaRegistry) Lineage(v SchemaRegistry) bool {
+	cur, hasMore := v.GetLatest()
+
+	for hasMore {
```

Comment:
maybe I'm missing something, but why do we need a loop here? Can we just check if `v.GetLatest().DeployID()` is in sr? If not, then SR does not not extend `v` right?

Oh I see its because you need to make sure the entire history of `v` is contained within `sr`...

Ok that makes sense although I feel like this would be a lot simpler if we just also kept track of the index in the history so we could just see if it came later in the list. Maintaining the sort order shouldn't be hard (and we can still keep the linked-list thing as well as an additional sanity thing)

Reply:
I tried to generate the index, did not seem to be simpler than this.

Reply:
ok fair